File: java/src/test/java/com/murex/PascalTriangleTest.java
/*
Copyright (c) 2021 Murex

Permission is hereby granted, free of charge, to any person obtaining a copy
of this software and associated documentation files (the "Software"), to deal
in the Software without restriction, including without limitation the rights
to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
copies of the Software, and to permit persons to whom the Software is
furnished to do so, subject to the following conditions:

The above copyright notice and this permission notice shall be included in all
copies or substantial portions of the Software.

THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
SOFTWARE.
*/

package com.murex;

import static org.junit.jupiter.api.Assertions.assertArrayEquals;
import org.junit.jupiter.api.Disabled;
import org.junit.jupiter.api.Test;

public class PascalTriangleTest {
    @Disabled
    @Test
    public void acceptance_test() {
        // /!\ Should only pass at the end:
        // - Keep ignored
        // - Write and pass intermediate tests
        // - Unignore at the end when should be passing

        assertArrayEquals(new int[]{1}, PascalTriangle.line(0));
        assertArrayEquals(new int[]{1, 1}, PascalTriangle.line(1));
        assertArrayEquals(new int[]{1, 2, 1}, PascalTriangle.line(2));
        assertArrayEquals(new int[]{1, 3, 3, 1}, PascalTriangle.line(3));
        assertArrayEquals(new int[]{1, 4, 6, 4, 1}, PascalTriangle.line(4));
        assertArrayEquals(new int[]{1, 5, 10, 10, 5, 1}, PascalTriangle.line(5));
        assertArrayEquals(new int[]{1, 6, 15, 20, 15, 6, 1}, PascalTriangle.line(6));
        assertArrayEquals(new int[]{1, 7, 21, 35, 35, 21, 7, 1}, PascalTriangle.line(7));
        
        //...

        assertArrayEquals(new int[]{1, 15, 105, 455, 1365, 3003, 5005, 6435, 6435, 5005, 3003, 1365, 455, 105, 15, 1}, PascalTriangle.line(15));
    }
}

File: python/tests/pascal_triangle_test.py
from slow_code.pascal_triangle import line
import pytest


class TestPascalTriangle:

    # /!\ Should only pass at the end:
    # - Keep skipped
    # - Write and pass intermediate tests
    # - Un-skip at the end when should be passing
    @pytest.mark.skip(reason="test currently disabled")
    def test_acceptance(self):
        assert line(0) == [1]
        assert line(1) == [1, 1]
        assert line(2) == [1, 2, 1]
        assert line(3) == [1, 3, 3, 1]
        assert line(4) == [1, 4, 6, 4, 1]
        assert line(5) == [1, 5, 10, 10, 5, 1]
        assert line(6) == [1, 6, 15, 20, 15, 6, 1]
        assert line(7) == [1, 7, 21, 35, 35, 21, 7, 1]
        # ...
        assert line(15) == [1, 15, 105, 455, 1365, 3003, 5005, 6435, 6435, 5005, 3003, 1365, 455, 105, 15, 1]
